File: Python/kodlar.py
# Student sinfi: bir tələbə obyekti üçün sahələr və metodlar
class Student:

    # Konstruktor: yeni Student obyekti yaratmaq üçün istifadə olunur
    def __init__(self, ad):
        self.__ad = ad  # sahəni konstruktor parametri ilə doldururuq
                        # __ -> private deməkdir (inkapsulyasiya)

    # Getter metodu: ad sahəsini oxumaq üçün
    def get_ad(self):
        return self.__ad

    # Setter metodu: ad sahəsini dəyişmək üçün
    def set_ad(self, ad):
        self.__ad = ad


# Metod obyekt qəbul edir və obyektin ad sahəsini dəyişdirir
def ad_deyis(s, yeni_ad):
    s.set_ad(yeni_ad)  # student obyektinin ad sahəsini dəyişdiririk


# Əsas hissə
if __name__ == "__main__":

    # Yeni Student obyekti yaradılır
    s1 = Student("Ravan")
    print("Əvvəl:", s1.get_ad())  # Getter vasitəsilə adı ekrana çıxarır

    # Metod vasitəsilə obyektin adını dəyişdiririk
    ad_deyis(s1, "Ismayil")
    print("Sonra:", s1.get_ad())  # Yenilənmiş ad ekrana çıxarılır




#-------------------------------------------------------------------------
# User sinfi - İstifadəçi məlumatlarını saxlamaq üçün
class User:
    def __init__(self, ad, ata_adi, email, telefon, yas):
        self.ad = ad
        self.ata_adi = ata_adi
        self.email = email
        self.telefon = telefon
        self.yas = yas

    # Məlumatları ekrana çap edən metod
    def print_info(self):
        print(
            f"Ad: {self.ad}, Ata adı: {self.ata_adi}, "
            f"Telefon: {self.telefon}, Email: {self.email}, Yaş: {self.yas}"
        )


users = []
say = 0


def neceneferin_daxil_edilmesi():
    global say
    while True:
        try:
            say = int(input("Neçə nəfər üçün məlumat daxil etmək istəyirsiniz: "))
            if say > 0:
                break
            else:
                print("Ədəd 0-dan böyük olmalıdır.")
        except ValueError:
            print("Zəhmət olmasa düzgün ədəd daxil edin.")


def daxil_et(mesaj, yoxlama):
    while True:
        daxil_edilen = input(f"{mesaj} daxil edin: ")
        if yoxlama(daxil_edilen):
            return daxil_edilen
        else:
            print("Yanlış daxil etdiniz, yenidən cəhd edin.")


def melumatlarin_daxil_edilmesi():
    for _ in range(say):
        ad = daxil_et("Ad", lambda x: x.isalpha())
        ata_adi = daxil_et("Ata adı", lambda x: x.isalpha())
        email = daxil_et("E-poçt ünvanı", lambda x: "@" in x and "." in x)
        telefon = int(
            daxil_et("Telefon nömrəsi", lambda x: x.isdigit() and len(x) == 10)
        )
        yas = int(
            daxil_et("Yaş", lambda x: x.isdigit() and int(x) > 0)
        )

        users.append(User(ad, ata_adi, email, telefon, yas))


def main():
    neceneferin_daxil_edilmesi()
    melumatlarin_daxil_edilmesi()

    print("\nDaxil edilən istifadəçilər:")
    for user in users:
        user.print_info()


if __name__ == "__main__":
    main()

File: java/Kodlar.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import java.util.function.Predicate;

// User sinfi - İstifadəçi məlumatlarını saxlamaq üçün
class User {
    private String ad;
    private String ataAdi;
    private String email;
    private long telefon;
    private int yas;

    // Konstruktor
    public User(String ad, String ataAdi, String email, long telefon, int yas) {
        this.ad = ad;
        this.ataAdi = ataAdi;
        this.email = email;
        this.telefon = telefon;
        this.yas = yas;
    }

    // Məlumatları çap edən metod
    public void print() {
        System.out.println(
            "Ad: " + ad +
            ", Ata adı: " + ataAdi +
            ", Telefon: " + telefon +
            ", Email: " + email +
            ", Yaş: " + yas
        );
    }
}

public class Main {
    static List<User> users = new ArrayList<>();
    static int say = 0;
    static Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {

        // 1. Neçə nəfər məlumat daxil ediləcəyini soruşuruq
        neceneferinDaxilEdilmesi();

        // 2. Məlumatları daxil edirik
        melumatlarinDaxilEdilmesi();

        // 3. Çap edirik
        System.out.println("\nDaxil edilən istifadəçilər:");
        for (User user : users) {
            user.print();
        }
    }

    static void neceneferinDaxilEdilmesi() {
        System.out.print("Neçə nəfər üçün məlumat daxil etmək istəyirsiniz: ");
        while (true) {
            try {
                say = Integer.parseInt(scanner.nextLine());
                if (say > 0) break;
            } catch (NumberFormatException e) {
                // ignore
            }
            System.out.print("Yenidən düzgün bir ədəd daxil edin: ");
        }
    }

    static void melumatlarinDaxilEdilmesi() {
        for (int i = 0; i < say; i++) {
            String ad = daxilEt("Ad",
                    input -> !input.isEmpty() && Character.isLetter(input.charAt(0)));

            String ataAdi = daxilEt("Ata adı",
                    input -> !input.isEmpty() && Character.isLetter(input.charAt(0)));

            String email = daxilEt("E-poçt ünvanı",
                    input -> input.contains("@") && input.contains("."));

            long telefon = Long.parseLong(
                    daxilEt("Telefon nömrəsi",
                            input -> input.matches("\\d{10}"))
            );

            int yas = Integer.parseInt(
                    daxilEt("Yaş",
                            input -> input.matches("\\d+") && Integer.parseInt(input) > 0)
            );

            users.add(new User(ad, ataAdi, email, telefon, yas));
        }
    }

    static String daxilEt(String mesaj, Predicate<String> yoxlama) {
        String daxilEdilen;
        do {
            System.out.print(mesaj + " daxil edin: ");
            daxilEdilen = scanner.nextLine();
        } while (!yoxlama.test(daxilEdilen));
        return daxilEdilen;
    }
}
